Userland/Libraries/LibVideo/VP9/Parser.cpp

```diff
@@ -28,8 +28,53 @@ Parser::~Parser()
 {
 }
 
+Vector<size_t> Parser::parse_superframe_sizes(Span<const u8> frame_data)
+{
+    if (frame_data.size() < 1)
+        return {};
+
+    // If the checks in steps 1 and 3 both pass, then the chunk is determined to contain a superframe and each
+    // frame in the superframe is passed to the decoding process in turn.
+    // Otherwise, the chunk is determined to not contain a superframe, and the whole chunk is passed to the
+    // decoding process.
+
+    // The decoder determines the presence of a superframe by:
+    // 1. parsing the final byte of the chunk and checking that the superframe_marker equals 0b110,
+
+    // Reading from span data will be quicker than spinning up a BitStream.
+    u8 superframe_byte = frame_data[frame_data.size() - 1];
+    u8 superframe_marker = superframe_byte & 0b1110'0000;
+    if (superframe_marker == 0b1100'0000) {
```

Comment:
You could move the comment `// 1. parsing the final byte of the chunk and checking that the superframe_marker equals 0b110,` to just above this if to make it clearer

Also the comment says you check for `0b110` but you check for `0b1100'0000` so maybe add a NOTE why this is different from the comment?

Reply:
Good points, I'll make those changes.